salt/_pillar/velum.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+'''
+A module that adds data to the Pillar structure retrieved by an http request made to Velum
+
+
+Configuring the VELUM ext_pillar
+====================================
+
+Set the following Salt config to setup Foreman as external pillar source:
+
+.. code-block:: json
+
+  ext_pillar:
+    - velum:
+        - url: https://velum.example.com/internal-api/v1/pillar?version=1
+        - ca_bundle: /some/bundle.crt
+        - username: username
+        - password: secret
+
+Module Documentation
+====================
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import logging
+
+# Import Salt libs
+import salt.ext.six as six
+
+import json
+import hashlib
+
+pillar_sha256 = None
+pillar_cache = None
+cached_username = None
+cached_password = None
+
+def ext_pillar(minion_id,
+               pillar,  # pylint: disable=W0613
+               url=None,
+               ca_bundle=None,
+               username=None,
+               password=None):
+    '''
+    Read pillar data from HTTP response.
+
+    :param url String to make request
+    :param ca_bundle Path to CA bundle
+    :param username Username for basic-auth
+    :param password Password for basic-auth
+    :returns dict with pillar data to add
+    :returns empty if error
+    '''
+
+    global pillar_sha256
+    global pillar_cache
+    global cached_username
+    global cached_password
+
+    # Set up logging
+    log = logging.getLogger(__name__)
+
+    if username is None:
+        if cached_username is None:
+            with open('/var/lib/misc/infra-secrets/velum-internal-api-username', 'r') as f:
+                cached_username = f.read().strip()
+        username = cached_username
+
+    if password is None:
+        if cached_password is None:
+            with open('/var/lib/misc/infra-secrets/velum-internal-api-password', 'r') as f:
```

Comment:
Just do double check, neither the `username` nor the `password` are specified into the `returner.conf` (see above). That means `/var/lib/misc/infra-secrets/velum-internal-api-password` and `/var/lib/misc/infra-secrets/velum-internal-api-username` are going to be used.

Their values are defined inside of PR https://github.com/kubic-project/caasp-container-manifests/pull/116/ but I cannot see them mounted into the salt container. Am I missing something?

Reply:
Yes, `returner.conf` is not used anymore for the pillar access (note that we will be able to remove `returner.conf` completely when salt does not access the database for anything).

The `salt-master` container mounts `/var/lib/misc/infra-secrets` as a whole (https://github.com/kubic-project/caasp-container-manifests/blob/master/public.yaml#L186-L188), so this files should be accessible.